gobblin-core/src/main/java/org/apache/gobblin/source/extractor/partition/Partitioner.java

```diff
@@ -322,6 +333,17 @@ protected long getLowWatermark(ExtractType extractType, WatermarkType watermarkT
         lowWatermark = this.getAppendLowWatermark(watermarkType, previousWatermark, deltaForNextWatermark);
       }
     }
+
+    if (isBootStrapWithFixedHighWatermark()) {
```

Comment:
This looks the same as the else block above, so it can be merged by changing the if to ` if (!isBootStrapWithFixedHighWatermark() && (this.isFullDump() || this.isWatermarkOverride())) {`